### DISCUSSION TITLE
More Human String Number Sorting

### DISCUSSION THREAD

**Discussion by carlosmbe** [specify/specify7#5796] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
In Trees, Values starting with 1 are treated as smaller than everything else. As a result, you can end up with `Sec 100` coming before `Sec 2`. And so on and so forth. 

The easy solution would be that the tree has a numbering system like 01...09, 10.... But then if, for reasons beyond me, they continue to grow towards Sec 100, 1,000, 10,000 then the prior naming pattern they used would become useless and the entire tree becomes a mess overtimes. Unless they edit the older ones to match the new naming scheme.

<img width="1779" alt="Screenshot 2023-05-09 at 2 35 50 PM" src="https://github.com/specify/specify7/assets/53784701/fe407e5e-7ec0-42a9-99ac-5342e838b7f9">

**Describe the solution you'd like**
Alpha Numeric Order in a human way
IE
|___Sec 1
          |___Sec 2
          |___Sec20
          |___Sec 100
          |___Sec 2000
          |___Sec 10,000

**Comment by melton-jason** [comment]
The issue is how computers widely sort strings, in an alphabetical, [lexicographical way](https://stackoverflow.com/a/45950665). 
Each character is compared to the corresponding part (i.e., the character at the same index) of the other string it's being compared to. 
Thus 10 is _less than_ 2 because it compares the characters at the first index `1` and `2` and determines that `1` is less than `2`. Thus _any_ string starting with 1 will _always_ be less than 2. 

I will agree, this is very confusing to those trying to use a Numeric ordering system where a Alphabetical ordering system is used. 
The proper (and potentially hardest to implement) solution is to let the user decide which ordering system to use wherever we can enforce an ordering system. 
But the values they are sorting can still contain string characters. So how would you _numerically_ compare `SEC 2T3` to `BLAB 4B2Z`? 
Should we throw an error in this case? Sort based on ASCII values? (which i'm pretty sure SQL does for _lexical comparisons and sorting_!)

From a user standpoint, the _best_ solution might be to break the string into sub-strings with specific rules. 
For instance, a Storage Name in the Cabinet rank _MUST_ start with CAB and be followed by numbers, and only the numbers will be sorted (and sorted numerically). 
As you might imagine, this might become very complex very quickly.

**Comment by maxpatiiuk** [comment]
there is a `sortFunction` on the front-end that sorts like humans rather than like computers - we could totally use front-end sorting using that function here instead of back-end sorting like it does right now.

at the same time, not sure how often people have numbers in species names - though there are other cases too where sortFunction is better (i.e ç vs c)
